### ISSUE TITLE
Ensure Checkbox Has Accessible Labels

### ISSUE THREAD

**Issue by hawkticehurst** [microsoft/vscode-webview-ui-toolkit#176]
### Describe the bug

Based on Storybook accessibility audits, must ensure that checkboxes have an accessible label.

### To reproduce

Look at the accessibility tab for the icon button story:

[https://microsoft.github.io/vscode-webview-ui-toolkit/?path=/story/library-checkbox--default](https://microsoft.github.io/vscode-webview-ui-toolkit/?path=/story/library-checkbox--default)

### Expected behavior

Checkboxes should have accessible labels.

### Screenshots

<img width="844" alt="Screen Shot 2021-08-16 at 11 49 33 AM" src="https://user-images.githubusercontent.com/39639992/129628027-dbd3ddfa-b322-4060-a0aa-370b2d28fd5b.png">

### Desktop (please complete the following information):

- OS Version: `macOS 11.5`
- Toolkit Version: `v0.6.2`
